File: JSBASIC/190_mode_array.js
// 190	190_mode_array.js	Find the mode (most frequently occurring element) of an array.

let arr = [1, 2, 3, 2, 4, 2, 5, 3];
let freq = {};
let maxFreq = 0, mode;
for (let i = 0; i < arr.length; i++) {
    freq[arr[i]] = (freq[arr[i]] || 0) + 1;
    if (freq[arr[i]] > maxFreq) { maxFreq = freq[arr[i]]; mode = arr[i]; }
}
console.log("Array:", arr);
console.log("Mode:", mode, "(appears", maxFreq, "times)");

File: JavaBasic/190_mode_array.java
// 190	190_mode_array.java	Find the mode (most frequently occurring element) of an array.
import java.util.*;
public class _190_mode_array {
    public static void main(String[] args) {
        int[] arr = {1,2,3,2,4,2,5,3};
        Map<Integer,Integer> freq = new HashMap<>();
        for (int x:arr) freq.put(x, freq.getOrDefault(x,0)+1);
        int mode=arr[0], maxF=0;
        for (Map.Entry<Integer,Integer> e:freq.entrySet()) if(e.getValue()>maxF){maxF=e.getValue();mode=e.getKey();}
        System.out.println("Mode: " + mode + " (appears " + maxF + " times)");
    }
}
